### PR TITLE
[2419] Add financial incentives to v2 subject serialiser

### PR DESCRIPTION
**Pull request by davidgisbey**
### Context

We need to display a course's financial incentives. 

### Changes proposed in this pull request

- Adds a subjects financial incentives to the V2 subject serialiser.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
https://github.com/DFE-Digital/search-and-compare-ui/blob/903e35ea800ccc4cf0b06a18a36e037ab813e229/shared/ViewFormatters/CourseExtensions.cs#L135-L158

**Comment by dankmitchell** [comment]
@defong we can tackle that madness in a follow up PR 😅

**Comment by defong** [comment]
That **_madness_** is what hold us altogether...
:stuck_out_tongue_winking_eye:

File: spec/serializers/api/v2/serializable_subject_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  it { should have_attribute(:subject_code).with_value(non_bursary_subject.subject_code) }
+
+
+  context "when a bursary subject" do
+    it { should have_attribute(:bursary_amount).with_value(nil) }
+    it { should have_attribute(:early_career_payments).with_value(nil) }
+    it { should have_attribute(:scholarship).with_value(nil) }
+  end
+
+  context "when a bursary subject" do
```
Comment (misaka) [inline spec/serializers/api/v2/serializable_subject_spec.rb]: This context and the one above is the same, what's the diff?

Comment (davidgisbey) [reply spec/serializers/api/v2/serializable_subject_spec.rb]: yea sorry just hadn't edited it to non-bursary subject

**Comment by davidgisbey** [comment]
[Add financial incentives to subjects/courses endpoint](https://trello.com/c/RjkOTVUa/2419-add-financial-incentives-to-subjects-courses-endpoint)

**Pull request merged**